install-devenv.sh

```diff
@@ -1,70 +1,198 @@
 #!/bin/bash
+# Manual test:
+#   export DEBUG=1
+#   export SNTY_DEVENV_REPO=file://$PWD
+#   export SNTY_DEVENV_BRANCH="$(git branch --show-current)"
+#   ./install-devenv.sh
+PATH=/usr/bin:/bin:/usr/sbin:/sbin
+#HERE="$(cd "$(dirname "$0")"; pwd)"
+ME="$(basename "$0")"
+
+## functions
+# let users see important commands
+show() {(set -x; "$@")}
+# check if a command exists
+has() { command -v "$@" >/dev/null; }
+# tell the user something
+info() { colorize "$ansi_teal" "$@"; }
+# warn the user
+warn() { colorize "$ansi_yellow" "$@"; }
+error() { colorize "$ansi_red" "$@"; exit 1; }
+yesno() { # ask a question
+  prompt="$1$ansi_green [y/n]$ansi_reset: "
+  while :; do
+    if [[ "${CI:-}" ]]; then
+      REPLY="yes"
+      echo -n "$prompt"
+      info "$REPLY"
+    else
+      read -r -p "$prompt"
+    fi
 
-if [[ "$(uname -s)" != Darwin ]]; then
-    echo "Only macs are supported for now."
-    exit 1
-fi
-
-branch=${1:-main}
-
-devenv_cache="${HOME}/.cache/sentry-devenv"
-devenv_root="${HOME}/.local/share/sentry-devenv"
-devenv_bin="${devenv_root}/bin"
-devenv_python_root="${devenv_root}/python"
-
-mkdir -p "$devenv_cache" "$devenv_python_root" "$devenv_bin"
-
-platform=x86_64
-sha256=47e1557d93a42585972772e82661047ca5f608293158acb2778dccf120eabb00
-
-case "$(uname -m)" in
+    case $REPLY in
+        [yY]*) return 0 ;;
+        [nN]*) return 1 ;;
+        *) warn "Unrecognized response.";;
+    esac
+  done
+}
+# show a colorized message
+colorize() {
+  color="$1"
+  shift 1
+  message="$*"
+  echo >&2 "$color$message$ansi_reset"
+}
+http-get() {(
+  # print the content of a (https) URL to stdout
+  url="$1"
+  if has curl; then
+    curl -q -sSLf "$url"
+  elif has wget; then
+    wget -nv -O- "$url"
+  else
+    error 'Please install `curl` or `wget` and try again'
+  fi
+)}
+
+constants() {
+  # a few color codes
+  ansi_yellow=$'\x1b[33m'
+  ansi_red=$'\x1b[31m'
+  ansi_green=$'\x1b[1;34m'
+  ansi_teal=$'\x1b[36m'
+  ansi_reset=$'\x1b[m'
+  # fancy "prompt" for xtrace log
+  export PS4="+ $ansi_green\$$ansi_reset "
+
+  if [[ "${DEBUG:-}" || "${SNTY_DEVENV_DEBUG:-}" ]]; then
+    set -x
+    colorize() { :; }  # show messages just once
+  fi
+
+  # default values for system environment variables
+  USER=$(id -un)
+  HOME=$(eval 'echo ~')
+  XDG_DATA_HOME="${XDG_DATA_HOME:-$HOME/.local/share}"
+  XDG_CACHE_HOME="${XDG_DATA_HOME:-$HOME/.cache}"
+  OSTYPE="${OSTYPE:-$(uname -s | tr -d '0-9.' | tr '[:upper:]' '[:lower:]')}"
```

Comment:
Please ignore the default value. For some reason I have `darwin22.0` already set and this errors later: `Unexpected platform; please contact <team-devenv@sentry.io>: (darwin22-arm64 -> aarch64-??)`.

Also can we add a `#discuss-dev-infra`? People are gonna prefer slack over email.